xapian-core/cluster/cluster.cc

```diff
@@ -369,8 +366,8 @@ Cluster::clear() {
     cluster_docs.clear();
 }
 
-Centroid
-Cluster::get_centroid() const {
+Centroid&
+Cluster::get_centroid() {
```

Comment:
Ah, umm. Does the centroid get modified? (A const function can return a const reference.)

If `Centroid` ends up being one of our PIMPL classes, you can just return the value as before.